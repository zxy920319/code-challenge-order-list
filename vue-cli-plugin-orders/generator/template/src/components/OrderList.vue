<template>
  <div v-if="computedOrders" class="orders-wrapper">
    <v-card
      v-for="(order, index) in computedOrders"
      :key="index"
      @click="() => $emit('click:order', order)"
    >
      <v-card-text>
        <div>{{ order.orderTime }}</div>
        <p class="display-1 text--primary">
          {{ order.restaurantName }}
        </p>
        <p>{{ order.status }}</p>
        <div class="text--primary">
          {{ order.deliveryAddress }}
        </div>
      </v-card-text>
      <v-card-actions>
        <div class="font-weight-black">
          {{ `${order.orderTotal} ${currency}` }}
        </div>
      </v-card-actions>
    </v-card>
  </div>
</template>

<script>
import { mapGetters } from "vuex";
export default {
  props: {
    currency: {
      required: true,
      type: String
    }
  },
  computed: {
    ...mapGetters(["orders"]),
    computedOrders() {
      let result = this.orders;
      <% if (orderNumber) { %>
      const orderNumber = <%= orderNumber %>;
      <% } else { _%>
        const orderNumber = 5;
      <%_ } _%>
      if(this.orders.length > orderNumber) {
        result = this.orders.slice(0, orderNumber);
      }
      return result;
    }
  }
};
</script>
