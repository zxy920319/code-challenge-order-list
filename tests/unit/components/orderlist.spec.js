import OrderList from "@/components/OrderList.vue";
import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import VueRouter from "vue-router";
import Vuetify from "vuetify";

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);
localVue.use(Vuetify);

let vuetify = new Vuetify();
const mockData = [
  {
    orderTime: "2019-07-15 16:44:06",
    restaurantName: "Big Burger Joint",
    orderTotal: "16.95",
    deliveryAddress: "Streetname 1",
    status: "In transit"
  },
  {
    orderTime: "2019-07-07 17:33:00",
    restaurantName: "Crepes & Dutch Mini Pancakes",
    orderTotal: "24.35",
    deliveryAddress: "Bussinessstreet 420",
    status: "Delivered"
  },
  {
    orderTime: "2019-07-04 15:48:23",
    restaurantName: "Waffles + smoothies",
    orderTotal: "12.45",
    deliveryAddress: "Streetname 1",
    status: "Delivered"
  },
  {
    orderTime: "2019-06-19 16:55:16",
    restaurantName: "New Bali",
    orderTotal: "34.95",
    deliveryAddress: "Streetname 1",
    status: "Delivered"
  },
  {
    orderTime: "2019-06-01 11:33:36",
    restaurantName: "Substation",
    orderTotal: "12.45",
    deliveryAddress: "Streetname 1",
    status: "Delivered"
  },
  {
    orderTime: "2019-06-01 11:43:36",
    restaurantName: "Substation",
    orderTotal: "12.45",
    deliveryAddress: "Streetname 2",
    status: "Delivered"
  }
];
let getters = {
  orders: mockData => {
    if (mockData.last5Orders) {
      return Object.values(mockData.last5Orders.last5Orders);
    } else return [];
  }
};
let store = new Vuex.Store({
  getters
});

const routes = [
  {
    path: "/",
    name: "Orders"
  },
  {
    path: "/order-detail",
    name: "OrderList"
  }
];

const router = new VueRouter({
  routes
});

let wrapper = null;
const makeWrapper = () => {
  wrapper = shallowMount(OrderList, {
    propsData: {
      currency: "€"
    },
    router,
    store,
    vuetify,
    localVue
  });
};

describe("OrderList", () => {
  it("should have prop currency", () => {
    makeWrapper();
    expect(wrapper.vm.currency).toBe("€");
  });

  it("should have computed order and equal or less than 5", () => {
    expect(wrapper.html()).toMatchSnapshot();
    const localThis = { orders: mockData };
    expect(
      OrderList.computed.computedOrders.call(localThis).length
    ).toBeLessThanOrEqual(5);
  });
});
