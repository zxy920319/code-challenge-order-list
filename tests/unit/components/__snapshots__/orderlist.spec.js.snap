// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetail should have computed order and equal or less than 5 1`] = `
<v-scale-transition-stub origin="top center 0" currency="â‚¬">
  <!---->
</v-scale-transition-stub>
`;

exports[`OrderList should have computed order and equal or less than 5 1`] = `<v-row-stub tag="div" align="center" justify="start" class="text-left"></v-row-stub>`;
